src/transaction.ts

```diff
@@ -1543,22 +1626,31 @@ export class Dml extends Snapshot {
       query = {sql: query} as ExecuteSqlRequest;
     }
 
-    this.run(
-      query,
-      (
-        err: null | grpc.ServiceError,
-        rows: Rows,
-        stats: spannerClient.spanner.v1.ResultSetStats
-      ) => {
-        let rowCount = 0;
-
-        if (stats && stats.rowCount) {
-          rowCount = Math.floor(stats[stats.rowCount] as number);
-        }
+    const q = {opts: this._observabilityOptions};
+    Object.assign(q, query);
```

Comment:
you can do both in single line.

Reply:
So I was following precedent in this library because I see so many uses of `Object.assign` instead of the inline copying
```shell
src/batch-transaction.ts:    const reqOpts = Object.assign({}, query, Snapshot.encodeParams(query));
src/batch-transaction.ts:        const reqOpts = Object.assign({}, options, {
src/codec.ts:  options = Object.assign(defaultOptions, options);
src/codec.ts:        return Object.assign({name}, getType(value));
src/common-grpc/service.ts:    const retryOpts = Object.assign(
src/common-grpc/service.ts:    const retryOpts = Object.assign(
src/common-grpc/service.ts:    reqOpts = Object.assign({}, reqOpts);
src/common-grpc/service.ts:        Object.assign(
src/database.ts:    this.queryOptions_ = Object.assign(
src/database.ts:      Object.assign({}, queryOptions),
src/database.ts:          transaction!.requestOptions = Object.assign(
src/database.ts:        transaction.requestOptions = Object.assign(
src/database.ts:      const reqOpts = Object.assign(
src/index.ts:    options = Object.assign(
src/partial-result-stream.ts:    this._options = Object.assign({maxResumeRetries: 20}, options);
src/session-pool.ts:    this.options = Object.assign({}, DEFAULTS, options);
src/transaction-runner.ts:    this.options = Object.assign(defaults, options);
src/transaction.ts:    this.queryOptions = Object.assign({}, queryOptions);
src/transaction.ts:    request = Object.assign({}, request);
src/transaction.ts:    const reqOpts: spannerClient.spanner.v1.IReadRequest = Object.assign(
src/transaction.ts:        reqOpts: Object.assign({}, reqOpts, {resumeToken}),
src/transaction.ts:    query = Object.assign({}, query) as ExecuteSqlRequest;
src/transaction.ts:    query.queryOptions = Object.assign(
src/transaction.ts:      Object.assign({}, this.queryOptions),
src/transaction.ts:      reqOpts = Object.assign(query, {
src/transaction.ts:    Object.assign(q, query);
src/transaction.ts:          reqOpts: Object.assign({}, reqOpts, {resumeToken}),
src/transaction.ts:      Object.assign(readOnly, options);
src/transaction.ts:    Object.assign(q, query);
src/transaction.ts:      const batchError: BatchUpdateError = Object.assign(error, {
src/transaction.ts:            batchUpdateError = Object.assign(err, {rowCounts});
src/transaction.ts:            batchUpdateError = Object.assign(error, {
src/transaction.ts:      reqOpts.requestOptions = Object.assign(
src/v1/database_admin_client.ts:    opts = Object.assign({servicePath, port, clientConfig, fallback}, opts);
src/v1/instance_admin_client.ts:    opts = Object.assign({servicePath, port, clientConfig, fallback}, opts);
src/v1/spanner_client.ts:    opts = Object.assign({servicePath, port, clientConfig, fallback}, opts);
src/v1/spanner_executor_proxy_client.ts:    opts = Object.assign({servicePath, port, clientConfig, fallback}, opts);
system-test/spanner.ts:          const record = Object.assign(entry, {
test/batch-transaction.ts:      const expectedQuery = Object.assign({sql: QUERY.sql}, fakeParams);
test/batch-transaction.ts:        Object.assign({[LEADER_AWARE_ROUTING_HEADER]: 'true'})
test/batch-transaction.ts:      const expectedQuery = Object.assign({}, {sql: query});
test/batch-transaction.ts:        Object.assign({[LEADER_AWARE_ROUTING_HEADER]: 'true'})
test/common/service-object.ts:const fakePfy = Object.assign({}, pfy, {
test/common/service.ts:const fakeUtil = Object.assign({}, util, {
test/common/service.ts:    Object.assign(GrpcService, GrpcServiceCached);
test/common/service.ts:      Object.assign(
test/common/service.ts:      Object.assign({grpc: fakeGrpc}, CONFIG),
test/common/service.ts:      const config = Object.assign({}, CONFIG, {customEndpoint: true});
test/common/service.ts:      const config: Config = Object.assign({}, CONFIG);
test/common/service.ts:      const fakeGrpcMetadata = Object.assign(
test/common/service.ts:        const reqOpts = Object.assign(
test/common/service.ts:      const originalReqOpts = Object.assign({}, reqOpts);
test/common/service.ts:                Object.assign(
test/database.ts:    DatabaseCached = Object.assign({}, Database);
test/database.ts:        Object.assign(
test/database.ts:      const expectedReqOpts = Object.assign(
test/database.ts:        const options = Object.assign({}, OPTIONS, {gaxOptions});
test/database.ts:          Object.assign(
test/database.ts:      } = Object.assign(
test/index.ts:      const expectedOptions = Object.assign(
test/index.ts:      const config = Object.assign({}, CONFIG, {displayName});
test/index.ts:      const config = Object.assign({}, CONFIG, {edition});
test/index.ts:      const config = Object.assign({}, CONFIG, {processingUnits});
test/index.ts:      const config = Object.assign({}, CONFIG, {nodeCount, processingUnits});
test/index.ts:      const cfg = Object.assign({}, CONFIG, {gaxOptions: {}});
test/index.ts:      const options = Object.assign({}, OPTIONS, {gaxOptions});
test/index.ts:      const options = Object.assign({}, OPTIONS, {
test/index.ts:      const config = Object.assign({}, CONFIG, {displayName});
test/index.ts:      const cfg = Object.assign({}, CONFIG, {gaxOptions: {}});
test/index.ts:      const options = Object.assign({}, OPTIONS, {gaxOptions});
test/index.ts:      const options = Object.assign({}, OPTIONS, {
test/index.ts:      const options = Object.assign({}, OPTIONS, {gaxOptions});
test/index.ts:      const options = Object.assign({}, OPTIONS, {
test/index.ts:      const options = Object.assign(
test/instance-config.ts:        Object.assign({}, {instanceConfig: METADATA}),
test/instance-config.ts:        Object.assign({}, {instanceConfig: METADATA}, {gaxOpts: gaxOptions}),
test/instance-config.ts:          Object.assign({}, {instanceConfig: METADATA})
test/instance.ts:      const options = Object.assign({}, OPTIONS, {
test/instance.ts:      const originalOptions = Object.assign({}, options);
test/instance.ts:      const options = Object.assign({}, OPTIONS, {gaxOptions: {}});
test/instance.ts:        const options = Object.assign({}, OPTIONS, {gaxOptions});
test/instance.ts:        const createOptions: {autoCreate?: {}} = Object.assign({}, OPTIONS);
test/instance.ts:      const options = Object.assign({}, OPTIONS, {gaxOptions});
test/instance.ts:      const options = Object.assign({}, OPTIONS, {
test/instance.ts:      const options = Object.assign({}, OPTIONS, {gaxOptions});
test/instance.ts:      const options = Object.assign({}, OPTIONS, {
test/mockserver/mockdatabaseadmin.ts:    return Object.assign(error, {
test/mockserver/mockdatabaseadmin.ts:    return Object.assign(error, {
test/mockserver/mockinstanceadmin.ts:    return Object.assign(error, {
test/mockserver/mockinstanceadmin.ts:    return Object.assign(error, {
test/mockserver/mockspanner.ts:  return Object.assign(error, {
test/mockserver/mockspanner.ts:    return Object.assign(error, {
test/mockserver/mockspanner.ts:    return Object.assign(error, {
test/mockserver/mockspanner.ts:    const error = Object.assign(new Error(`Transaction aborted: ${name}`), {
test/mockserver/mockspanner.ts:    return Object.assign(error, {
test/partial-result-stream.ts:      const fakeResponse = Object.assign({}, RESULT, {values: []});
test/partial-result-stream.ts:    const RESULT_WITH_TOKEN = Object.assign({}, RESULT, {
test/session-pool.ts:    return Object.assign(new Session(DATABASE, name), props, {
test/session.ts:          Object.assign(
test/session.ts:          Object.assign(
test/spanner.ts:  const fooNotFoundErr = Object.assign(new Error('Table FOO not found'), {
test/transaction.ts:        const response = Object.assign({readTimestamp}, BEGIN_RESPONSE);
test/transaction.ts:        const query = Object.assign({}, QUERY, {
test/transaction.ts:        const fakeQuery = Object.assign({}, QUERY, {
test/transaction.ts:        const fakeQuery = Object.assign({gaxOptions: {}}, QUERY);
test/transaction.ts:        const fakeQuery = Object.assign({}, QUERY, expectedOptions);
test/transaction.ts:        const fakeQuery = Object.assign({}, QUERY, {
test/transaction.ts:        const fakeQuery = Object.assign({}, QUERY, {
test/transaction.ts:        const fakeQuery = Object.assign({}, QUERY, {
test/transaction.ts:        const fakeQuery = Object.assign({}, QUERY, {
test/transaction.ts:          Object.assign(
test/transaction.ts:          Object.assign(
test/transaction.ts:          Object.assign(
test/transaction.ts:          Object.assign(
test/transaction.ts:        const tableNotFoundErr = Object.assign(
test/transaction.ts:        const failedPreconditionErr = Object.assign(
test/transaction.ts:        const failedPreconditionErr = Object.assign(
test/transaction.ts:        const failedPreconditionErr = Object.assign(
test/transaction.ts:        const failedPreconditionErr = Object.assign(
test/transaction.ts:          Object.assign(
test/transaction.ts:            Object.assign(
test/transaction.ts:          Object.assign(
test/transaction.ts:          Object.assign(
```
but sure I can def change it

Reply:
But in these cases objects were already created and we were extending on the object using `Object.assign` . For eg, if we define object `observabilityOptions` in the constructor and we extend this object with the `query` here using `Object.assign`

Reply:
So that's what we are doing, we are copying the already create query and extending it then passing it into the tracer. Anyways I was just raising this as in the past such got mentioned but really the precedent was ignored. I'll get it inlined.

Reply:
Done, thanks!